text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,671 @@
+
+```
+[[entries]]
+mixin = "libpq"
+```
+
+A stack buildpack that needs to install mixins must select them from the build plan.
+
+### Example: Any Mixin Buildpack
+
+As an example of these resolution rules, consider an Apt stackpack that provide any (`*`) mixin. During the detect phase, the stackpack will pass in the following cases:
+* a mixin is required by a buildpack
+* a mixin is [required by an app](https://github.com/buildpacks/rfcs/pull/112)
+
+The example stackpack will be skipped after detection in the following cases:
+* no mixins are required by any component of the build
+* A mixin is required by a buildpack, but the stack already provides that mixin.
+* A mixin is required by an app, but the stack already provides that mixin.
+
+### Example: PostgreSQL Certs
+
+As another example of these resolution rules, consider a stackpack that provides the `libpq` mixin and a build plan dependency called `db-cert`. During the detect phase, the stackpack will pass in the following cases:
+* a Ruby app requires the `libpq` mixin and the `db-cert`
+* a Java app requires only the `db-cert`
+* a Python app requires the `db-cert` and the stack already provides the `libpq` mixin
+
+The example stackpack will be skipped after detection in the following cases:
+* an app requires the `libpq` mixin but not the `db-cert`
+* an app requires neither the `libpq` mixin nor the `db-cert`
+
+### Example: Chrome and Chromedriver
+
+As another example of these resolution rules, consider a stackpack that provides the `google-chrome-stable` mixin and a build plan dependency called `chromedriver`. During the detect phase, the stackpack will pass detection in the following cases:
+* an app requires the `google-chrome-stable` mixin and the `chromedriver`
+* an app requires only the `chromedriver`
+* an app requires the `chromedriver` and the stack already provides the `google-chrome-stable` mixin
+
+The example stackpack will be skipped after detection in the following cases:
+* an app requires the `google-chrome-stable` mixin but not the `chromedriver`
+* an app requires neither the `google-chrome-stable` mixin nor the `chromedriver`
+
+During the build phase the example stackpack will be given a [buildpack plan](https://github.com/buildpacks/spec/blob/main/buildpack.md#buildpack-plan-toml) containing the entries it must provide. For example, if both dependencies are required it will receive:
+
+```toml
+[[entries]]
+mixin = "google-chrome-stable"
+
+[[entries]]
+name = "chromedriver"
+```
+
+If only the `chromedriver` is required, or the `google-chrome-stable` is already provided by the stack, the build phase would receive:
+
+```toml
+[[entries]]
+name = "chromedriver"
+```
+
+If the stack does not already provide `google-chrome-stable`, the stackpack may install it anyways. This prevents the case where an app may not work at runtime if that app requires only the `chromedriver` and the `google-chrome-stable` is not provided by the stack.
```

Comment:
I'm not sure I quite follow this paragraph. The intro above says:

> If only the `chromedriver` is required, or the `google-chrome-stable` is already provided by the stack,

...but I'm not clear which scenario of the two it's referring to.

Does it mean something like this?

```suggestion
If the stack does not already provide `google-chrome-stable`, the stackpack may install it anyways, even if the app didn't require it. This prevents the case where an app may not work at runtime if that app requires only the `chromedriver` and the `google-chrome-stable` is not provided by the stack.
```